### DISCUSSION TITLE
Caching ReactiveList results

### DISCUSSION THREAD

**Discussion by shahchiragh** [appbaseio/reactivesearch#1596] [Ideas]
Greetings! I've implemented ReactiveSearch Vue UI library with Elasticsearch backend on one of my projects and I find it superb and amazing. With few learnings and client interactions, I figured it would be useful to cache results for the first time loads (before any query change) to save some loading time whenever an end user visits a search page.

Current situation: with about 10,000+ records the load time for reactivesearch is approx. ~ 10ms and goes less sometime too. And, I reload the same page a couple of times the time variation to render results (default) varies from ~5ms to 22ms. 

I understand that every time with a reload it would make a new request but that's where I want to pitch in the idea of caching.

Question: If the query has not changed for the reactivelist (default), could we cache the results response with a keep-alive property until the query changes?  I believe this would significantly reduce down the results loading time with every request made. 

I am curios to know if someone has thought of this before or has worked on similar aspect. I appreciate any thoughts or direction to achieve this. 

Thanks.

**Comment by siddharthlatest** [comment]
@shahchiragh This is a really interesting idea, thanks for bringing it up 👍 

We need to think more how similar client libraries handle caching and the possible side-effects. I can see this being very useful for scenarios where the underlying search data doesn't change very frequently. If we do this, it should be opt-in: i.e. a user has to specify the cache timeframe with a `ttl` like prop that can be present at the top level.

**Reply by siddharthlatest** [reply]
@shahchiragh 

[Cache Web API](https://developer.mozilla.org/en-US/docs/Web/API/Cache) is perhaps a good way to achieve this. The easiest way is to indiscriminately cache responses for a specified TTL.

We now also support built-in caching of results with [appbase.io cache](https://www.appbase.io/product/cache). This works on the server side, so it can effectively cache common patterns across the user base. The other advantage is it is easy to additionally have a configurable TTL and the ability to invalidate cache.
